### PR TITLE
GraphQL page improvements, and more

### PR DESCRIPTION
**Pull request by dmitryyankowski**
This PR does the following:

- Saves the schema and response to the local state, so if you refresh or navigate to a new page and back, it will show up again. https://github.com/liyasthomas/postwoman/pull/602/commits/21c6c07b3970252f10e3a5269e8372991b7dd2ca
- Update the `Run Query` button so it matches the `Get schema` button
image
- Running a query now shows "Loading..." in the response section, similar to when you click "Get Schema" https://github.com/liyasthomas/postwoman/pull/602/commits/ab49a125b290b760c3a8c70bddd9b94c9eb566a5
- Saves collapsed sections to the local state, similarly to the schema and response, Postwoman will show sections collapsed if you refresh/navigate away. (This will work on all pages; Not just the GraphQL page, as it uses the route as a key) https://github.com/liyasthomas/postwoman/pull/602/commits/2d5b70ab8511c3ee46a8fb76f1a3aec386f5e376
- Fixes auto scroll toggle (While I don't necessarily agree with how settings are initialized in the store; I've added checks for undefined auto scroll keys) https://github.com/liyasthomas/postwoman/pull/602/commits/6d534d312f06ead99d932072fe677194cbfb2963
- (Fix) Set `showExtensionsToast` to false in localStorage when no is clicked on the extensions toast https://github.com/liyasthomas/postwoman/pull/602/commits/1d616c7ec6bc6149edb17d8e94592c114e3d33e2
- Disable spell check on input fields (Shows red line under text if spell check is enabled)
- (Fix) Error message for failed queries/mutations on GraphQL page
- (Fix) Disable json linting for response editor on GraphQL page https://github.com/liyasthomas/postwoman/pull/605#issuecomment-591197382
- Moved `.prettierrc` and `.huskyrc` contents to the `package.json` file, so we have 2 less files to manage https://github.com/liyasthomas/postwoman/pull/602/commits/50339a2480205cd46c866aa314cb57d8d689b9c0

**Note**: I haven't gotten around to saving the types/queries/mutation to the local state yet (the Docs) But I will get to that soon :)

### CODE REVIEW

**Comment by liyasthomas** [comment]
@dmitryyankowski pre-commit hooks have been introduced hence please rebase to current master and pull latest change, then send PR.

**Comment by dmitryyankowski** [comment]
> @dmitryyankowski pre-commit hooks have been introduced hence please rebase to current master and pull latest change, then send PR.

Done 🚀 Just waiting for tests to pass :)

File: store/mutations.js [hunk 1]
```diff
@@ -7,6 +7,18 @@ export default {
...
+    theme.collapsedSections.includes(value)
+      ? (theme.collapsedSections = theme.collapsedSections.filter(section => section !== value))
+      : theme.collapsedSections.push(value)
+  },
+
+  setCollapsedSection({ theme }, value) {
+    theme.collapsedSections.includes(value)
+      ? (theme.collapsedSections = theme.collapsedSections.filter(section => section !== value))
+      : theme.collapsedSections.push(value)
+  },
```
Comment (liyasthomas) [inline store/mutations.js]: Duplicate code: This property is overwritten by another property in the same object literal.

https://lgtm.com/projects/g/liyasthomas/postwoman/rev/pr-b909ed91495c8b2ba56a610dcadf058045bea26b

Comment (dmitryyankowski) [reply store/mutations.js]: What the heck haha. Not sure how that duplicate got there. Fixed :)

**Comment by AndrewBastin** [comment]
Pro-tip for future PRs: If your PR is in work in progress, open it as a Draft PR.

**Comment by dmitryyankowski** [comment]
> Pro-tip for future PRs: If your PR is in work in progress, open it as a Draft PR.

Nice. Didn't even know what Draft PR's were until now 😄 Thanks for the tip!

**Comment by dmitryyankowski** [comment]
@liyasthomas LGTM? 🚀

**Comment by liyasthomas** [comment]
i was waiting you to remove WIP from title, so that I could review this. Is this finished?

**Comment by dmitryyankowski** [comment]
> i was waiting you to remove WIP from title, so that I could review this. Is this finished?

Yep finished (for now). I have more updates and improvements i'd like to make; but as @AndrewBastin suggested, i'll start that in a new Draft PR :)

**Comment by liyasthomas** [comment]
Sounds good. Will merge ASAP after review from @AndrewBastin

**Comment by dmitryyankowski** [comment]
Please hold off on the final review @AndrewBastin Need to fix some issues that the new json linter introduced :) Will @ you when i'm ready!

**Comment by dmitryyankowski** [comment]
@liyasthomas can we address the build error before I push any more changes? Getting `CypressError: Timed out retrying: expected '<div#response-details-wrapper>' to contain 'FAKE Cat API'` again in the build step. 😭

**Comment by liyasthomas** [comment]
@dmitryyankowski I don't know exactly what's wrong with the CI, locally everything seems to be working fine.

**Comment by dmitryyankowski** [comment]
> @dmitryyankowski I don't know exactly what's wrong with the CI, locally everything seems to be working fine.

Yeah it always works fine locally for me too. It does something weird in Travis CI, and times out for some reason 😭

**Comment by liyasthomas** [comment]
Lemme bump dependencies and see whether is there any issue with the Cypress itself. Sorry for the inconveniences.

**Comment by dmitryyankowski** [comment]
@liyasthomas This is my final commit for this PR. @AndrewBastin can you review this PR? Thank you!

**Pull request merged**